### DISCUSSION TITLE
Make u (username) parameter optional on user related endpoints and use authenticated user as default

### DISCUSSION THREAD

**Discussion by gastongaiduk** [RetroAchievements/RAWeb#2906] [Ideas]
Hello guys, I've been wondering about the fact that there's no possible way to know which user I am by just using the API key (I may be missing something).

Not sure if this is intended, couldn't find any conversation regarding this, but I do see sense on not having to ask for user name when we already have the user api key and we just want to fetch our authenticated user information.

My suggestion would be to use the authenticated user by default on all user related endpoints where we expect the u parameter.

What do you think?

**Comment by wescopeland** [comment]
cc @ioslife

**Comment by gastongaiduk** [comment]
I was thinking on something like [this](https://github.com/RetroAchievements/RAWeb/compare/master...gastongaiduk:retroachievements-web:api-make-u-parameter-optional-on-user-related-endpoints), for example on the API_GetUserProfile endpoint.

Of course I'll work on changing the docs as well if we proceed.

**Reply by wescopeland** [reply]
@Jamiras Do you have any opinion?

**Reply by Jamiras** [reply]
Feels like a security issue to me. If the user has an API key, but doesn't know which user it's associated to, this provides a way to make that connection.

As this particular API key is only valid for the public APIs, and I can't think of anything the user could do with the associated username that they couldn't do with just the API key, it's probably safe.

At some point in the past, we required both the username and API key for authentication. I don't recall when or why we switched to just requiring the API key.

@luchaos - do you have any concerns about this?

**Reply by wescopeland** [reply]
> Feels like a security issue to me. If the user has an API key, but doesn't know which user it's associated to, this provides a way to make that connection.

I hadn't considered this, but for this reason alone I would be opposed to this change.

If we go down this path, it seems like we essentially have to guarantee that the web API remains read-only indefinitely, and I'd prefer to have us not paint ourselves into that corner.

**Reply by gastongaiduk** [reply]
Yeah it does make sense, perhaps is not a bad idea for when we have a 2FA or any other sort of key that requires a pair to work (I'm thinking on something like AWS does with the client id and the client secret).

Going to close this one and keep an eye on new updates to come with a new discussion regarding this topic if needed.

Thanks for your thoughts @wescopeland @Jamiras
